iphone Xr on iOS13 returns unknown architecture

When running `flutter run` with asserts enabled, I get an assert in the method [getIOSArchForName()](https://github.com/flutter/flutter/blob/2d642e95e129b32ed45f799e1b9aad1e6f78cbcf/packages/flutter_tools/lib/src/build_info.dart#L319) when the arch is `arm64e`.

## Steps to Reproduce

Enable asserts for the flutter tool, and run `flutter run -v` on an iPhone Xr with iOS13.1 installed:

Flutter crash report; please file at https://github.com/flutter/flutter/issues.

## command

flutter run -v

## exception

_AssertionError: 'package:flutter_tools/src/build_info.dart': Failed assertion: line 320 pos 10: 'false': is not true.

```
#0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:40:39)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:36:5)
#2      getIOSArchForName (package:flutter_tools/src/build_info.dart:320:3)
#3      IOSDevice.startApp (package:flutter_tools/src/ios/devices.dart:276:34)
<asynchronous suspension>
#4      FlutterDevice.runHot (package:flutter_tools/src/resident_runner.dart:378:54)
<asynchronous suspension>
#5      HotRunner.run (package:flutter_tools/src/run_hot.dart:259:39)
<asynchronous suspension>
#6      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:481:37)
<asynchronous suspension>
#7      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:557:18)
<asynchronous suspension>
#8      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:457:33)
<asynchronous suspension>
#9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:157:29)
<asynchronous suspension>
#10     _rootRun (dart:async/zone.dart:1124:13)
#11     _CustomZone.run (dart:async/zone.dart:1021:19)
#12     _runZoned (dart:async/zone.dart:1516:10)
#13     runZoned (dart:async/zone.dart:1463:12)
#14     AppContext.run (package:flutter_tools/src/base/context.dart:156:18)
<asynchronous suspension>
#15     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:446:20)
#16     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
<asynchronous suspension>
#17     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:416:21)
<asynchronous suspension>
#18     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:157:29)
<asynchronous suspension>
#19     _rootRun (dart:async/zone.dart:1124:13)
#20     _CustomZone.run (dart:async/zone.dart:1021:19)
#21     _runZoned (dart:async/zone.dart:1516:10)
#22     runZoned (dart:async/zone.dart:1463:12)
#23     AppContext.run (package:flutter_tools/src/base/context.dart:156:18)
<asynchronous suspension>
#24     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:367:19)
<asynchronous suspension>
#25     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#26     new Future.sync (dart:async/future.dart:224:31)
#27     CommandRunner.run (package:args/command_runner.dart:112:14)
#28     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:251:18)
#29     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:63:22)
<asynchronous suspension>
#30     _rootRun (dart:async/zone.dart:1124:13)
#31     _CustomZone.run (dart:async/zone.dart:1021:19)
#32     _runZoned (dart:async/zone.dart:1516:10)
#33     runZoned (dart:async/zone.dart:1500:12)
#34     run.<anonymous closure> (package:flutter_tools/runner.dart:61:18)
<asynchronous suspension>
#35     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:157:29)
<asynchronous suspension>
#36     _rootRun (dart:async/zone.dart:1124:13)
#37     _CustomZone.run (dart:async/zone.dart:1021:19)
#38     _runZoned (dart:async/zone.dart:1516:10)
#39     runZoned (dart:async/zone.dart:1463:12)
#40     AppContext.run (package:flutter_tools/src/base/context.dart:156:18)
<asynchronous suspension>
#41     runInContext (package:flutter_tools/src/context_runner.dart:63:24)
<asynchronous suspension>
#42     run (package:flutter_tools/runner.dart:50:10)
#43     main (package:flutter_tools/executable.dart:65:9)
<asynchronous suspension>
#44     main (file:///Users/fujino/git/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#45     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:303:32)
#46     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
```

## flutter doctor

```
[32m[✓][39m Flutter (Channel stop-leaking-iproxy, v1.10.13-pre.8, on Mac OS X 10.14.6 18G87, locale en-US)
    [32m•[39m Flutter version 1.10.13-pre.8 at /Users/fujino/git/flutter
    [32m•[39m Framework revision b3f8062a06 (2 hours ago), 2019-10-07 09:32:09 -0700
    [32m•[39m Engine revision 7d90779bb6
    [32m•[39m Dart version 2.6.0 (build 2.6.0-dev.0.0 d6c6d12ebf)

[32m[✓][39m Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    [32m•[39m Android SDK at /Users/fujino/Library/Android/sdk
    [32m•[39m Android NDK location not configured (optional; useful for native profiling support)
    [32m•[39m Platform android-29, build-tools 28.0.3
    [32m•[39m Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    [32m•[39m Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    [32m•[39m All Android licenses accepted.

[32m[✓][39m Xcode - develop for iOS and macOS (Xcode 11.0)
    [32m•[39m Xcode at /Users/fujino/Downloads/./Xcode.app/Contents/Developer
    [32m•[39m Xcode 11.0, Build version 11A419c
    [32m•[39m CocoaPods version 1.7.5

[32m[✓][39m Android Studio (version 3.4)
    [32m•[39m Android Studio at /Applications/Android Studio.app/Contents
    [32m•[39m Flutter plugin version 37.0.1
    [32m•[39m Dart plugin version 183.6270
    [32m•[39m Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[32m[✓][39m Connected device (1 available)
    [32m•[39m Christopher’s iPhone • 00008020-00017DA80CC1002E • ios • iOS 13.1

[32m•[39m No issues found!
```

--- a/packages/flutter_tools/lib/src/base/build.dart
+++ b/packages/flutter_tools/lib/src/base/build.dart
@@ -142,7 +142,7 @@ class AOTSnapshotter {
       genSnapshotArgs.add('--no-use-integer-division');
     }
 
-    // The name of the debug file must contain additonal information about
+    // The name of the debug file must contain additional information about
     // the architecture, since a single build command may produce
     // multiple debug files.
     final String archName = getNameForTargetPlatform(platform, darwinArch: darwinArch);

--- a/packages/flutter_tools/lib/src/build_info.dart
+++ b/packages/flutter_tools/lib/src/build_info.dart
@@ -390,6 +390,7 @@ DarwinArch getIOSArchForName(String arch) {
     case 'armv7':
       return DarwinArch.armv7;
     case 'arm64':
+    case 'arm64e': // iPhone XS/XS Max/XR and higher. arm64 runs on arm64e devices.
       return DarwinArch.arm64;
     case 'x86_64':
       return DarwinArch.x86_64;

--- a/packages/flutter_tools/test/general.shard/build_info_test.dart
+++ b/packages/flutter_tools/test/general.shard/build_info_test.dart
@@ -85,4 +85,12 @@ void main() {
     expect(getNameForTargetPlatform(TargetPlatform.ios, darwinArch: DarwinArch.x86_64), 'ios-x86_64');
     expect(getNameForTargetPlatform(TargetPlatform.android), isNot(contains('ios')));
   });
+
+  test('getIOSArchForName on Darwin arches', () {
+    expect(getIOSArchForName('armv7'), DarwinArch.armv7);
+    expect(getIOSArchForName('arm64'), DarwinArch.arm64);
+    expect(getIOSArchForName('arm64e'), DarwinArch.arm64);
+    expect(getIOSArchForName('x86_64'), DarwinArch.x86_64);
+    expect(() => getIOSArchForName('bogus'), throwsAssertionError);
+  });
 }

